src/index.js

```diff
@@ -53,9 +53,40 @@ function isReactClass(superClass, scope, globalOptions) {
   return answer
 }
 
+function areSetsEqual(set1, set2) {
+  if (set1.size !== set2.size) {
+    return false
+  }
+
+  return !Array.from(set1).some(item => !set2.has(item))
+}
+
 export default function(api) {
   const { template, types, traverse } = api
 
+  const nestedIdentifiers = new Set()
+  const removedPaths = new Set()
+  const collectNestedIdentifiers = {
+    Identifier(path) {
+      if (path.parent.type === 'MemberExpression') {
+        return
+      }
+
+      if (
+        path.parent.type === 'ObjectProperty' &&
+        (path.parent.key === path.node || path.parent.shorthand)
```

Comment:
Do we need to check if `path.parent.computed`?

```js
{ [computedKey]: 'foo' }
```

Reply:
No, I think we still want identifiers in computed object keys, since by removing the object we may no longer need the identifier used in the computed key and we will want to clean those up.

Reply:
But I think we might not actually need the next check for CallExpressions. I'll remove that now.